broadcast/strategy.go

```diff
@@ -11,31 +12,33 @@ const (
 	OneByOneStrategy StrategyName = "OneByOneStrategy"
 )
 
-type ExecutionFunc func(context.Context, []func(context.Context) error) error
+type Result interface{}
+
+type ExecutionFunc func(context.Context) (Result, error)
+type StrategyExecutionFunc func(context.Context, []ExecutionFunc) (Result, error)
 
 type Strategy struct {
 	name          StrategyName
-	executionFunc ExecutionFunc
+	executionFunc StrategyExecutionFunc
 }
 
-func New(name StrategyName, executionFunc ExecutionFunc) *Strategy {
+func New(name StrategyName, executionFunc StrategyExecutionFunc) *Strategy {
 	return &Strategy{name: name, executionFunc: executionFunc}
 }
 
-func (s *Strategy) Execute(ctx context.Context, executionFuncs []func(context.Context) error) error {
+func (s *Strategy) Execute(ctx context.Context, executionFuncs []ExecutionFunc) (Result, error) {
 	return s.executionFunc(ctx, executionFuncs)
 }
 
 var (
-	OneByOne = New(OneByOneStrategy, func(ctx context.Context, executionFuncs []func(context.Context) error) error {
+	OneByOne = New(OneByOneStrategy, func(ctx context.Context, executionFuncs []ExecutionFunc) (Result, error) {
```

Comment:
now, I'm not sure if this is best approach to create this strategy here.
Maybe it would be better to expose only names (as const) and factory function,
and factory function would return appropriate strategy for name